examples/error-subtyping/error_subtyping.bal

```diff
@@ -1,21 +1,53 @@
 import ballerina/io;
 
-// `distinct` creates a new subtype.
-type XErr distinct error;
-type YErr distinct error;
+type InvalidIntDetail record {|
+    int value;
+|};
 
-type Err XErr|YErr;
+// Subtype of `InvalidIntDetail`.
+type InvalidI32Detail record {|
+    int:Signed32 value;
+|};
 
-// The name of the distinct type can be used with the `error` constructor to create an error value 
-// of that type. `err` holds an `error` value of type `XErr`.
-Err err = error XErr("Whoops!");
+// Error with the `InvalidIntDetail` type as the detail type.
+type InvalidIntError error<InvalidIntDetail>;
 
-function desc(Err err) returns string {
-    // The `is` operator can be used to distinguish distinct subtypes.
-    return err is XErr ? "X" : "Y";
+// `error` with `InvalidI32Detail` as the detail type. Thus it is a subtype of `InvalidIntError`.
+type InvalidI32Error error<InvalidI32Detail>;
 
-}
+// Distinct error with the `InvalidIntDetail` type as the detail type and a unique type ID.
+// Therefore, this is a proper subtype of `InvalidIntError`, but doesn't have a subtype relationship 
+// with `AnotherDistinctIntError` because they have different type IDs.
+type DistinctIntError distinct error<InvalidIntDetail>;
+
+// Another `error` with `InvalidIntDetail` as the detail type and different type ID to `DistinctIntError`
```

Comment:
```suggestion
// Another distinct error with `InvalidIntDetail` as the detail type, but a different type ID to `DistinctIntError`
```

id vs ID is not consistent across examples.

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-distribution/commit/c19adcc407f68b124fae5f39fe8a901d22566377